### DISCUSSION TITLE
Blog #7

### DISCUSSION THREAD

**Discussion by Unleqit** [AlinaBoess/SoftwareEngineeringProjektTINF23B5#7] [General]
Siebter Blogeintrag - Architecture Significant Requirements 

Schön, dass ihr es wieder zu uns geschafft habt! 😊 Diese Woche stand bei uns die Definition der Architecture Significant Requirements im Fokus:

Nachfolgend möchten wir euch gerne unsere Architecture Significant Requirements auf Grundlage des 3-Schritte-Ansatzes, sowie die dazugehörigen Gründe präsentieren; eine Zusammenfassung kann [hier](https://github.com/AlinaBoess/SoftwareEngineeringProjektTINF23B5/blob/main/documentation/Aufgaben/Aufgabe%208/ARS.md) eingesehen werden.

1. Usability: Ein **Responsive Design** , sowie klare Fehlermeldungen werden implementiert, um die Benutzerfreundlichkeit zu gewährleisten.
2. Performance: Wir haben Wert auf die **Optimierung der Datenbankzugriffe**, sowie die Verwendung von Caching gelegt, um Antwortzeiten zu minimieren und unnötige serverseitige Last zu verhindern.
3. Sicherheit: Es werden eine TLS-Verschlüsselung für Datenübertragungen und rollenbasierte Zugriffsrechte für Benutzer und Administratoren eingeführt, um die Sicherheit und Vertraulichkeit der Daten zu gewährleisten und unerlaubte Modifikationen am System zu unterbinden.
4. Skalierbarkeit: Der Einsatz einer Cloud-Hosting-Lösung wurde beschlossen, um bei Spitzenlasten zusätzliche Serverkapazitäten bereitzustellen.
5. Datenbankstruktur: Die Beziehungstabellen (z. B. für Benutzer, Tische, und Reservierungen) sind optimiert für schnelle Abfragen und Erweiterbarkeit, um das System problemlos auf eine höhere Nutzerzahl skalieren zu können.
6. Klassendesign: Die Trennung von Benutzerrollen wie Admin und User unterstützt die Wartbarkeit und Erweiterbarkeit des Systems und dessen einzelnen Komponenten.
7. Scrum-Setup: Iterative Entwicklungszyklen mit Reviews und Retrospektiven fördern die kontinuierliche Verbesserung der Architektur, weswegen wir ein projektbegleitendes Jira führen, um solche Aufgaben effizient im Team zu delegieren.

Des Weiteren findet ihr [hier](https://github.com/AlinaBoess/SoftwareEngineeringProjektTINF23B5/blob/main/documentation/Aufgaben/Aufgabe%208/utilityTree.md) unseren Utility Tree, welcher 5 Szenarien für Qualitätsattribute beinhaltet, auf welche wir bei der Entwicklung unseres Projektes besonderen Wert legen.

Wir hoffen, dass die oben erläuterten Entscheidungen schlüssig und gut nachvollziehbar sind; schaut gerne nächste Woche wieder vorbei und lasst uns wissen, was ihr denkt, oder falls ihr weiterführende Fragen zu unseren ASR habt. 😊

Liebe Grüße
Lukas, Alex, Moumen, Yahya und Alina

**Comment by lucariebel** [comment]
Hi zusammen,

aus eurem ASR ist gut ersichtlich, was das Ziel eurer Anwendung sein soll. Schnell und zuverlässig und dabei immer darauf zu achten, dass es skalierbar bleibt.

Viele Grüße 
Euer WebShop24

**Comment by GermanJesus-lul** [comment]
Hey,

Eure ASR sind gut und sinngemäß. Vor allem finde ich gut, dass ihr die _interaction cost_ dabei im Blick habt. Wie habt ihr vor die 2-Faktor-Authentifizierung umzusetzen?

Mit freundlichen Grüßen,
Julius und das Rezeptbuch Team

**Comment by Paula642** [comment]
Hallo,

Die Fokuspunkte auf Usability, Performance, Sicherheit und Skalierbarkeit sind klar erkennbar und machen das System wirklich benutzerfreundlich und robust. Besonders die Maßnahmen zur Datenbankoptimierung und zur Cloud-Skalierung wirken zukunftssicher und gut durchdacht. Auch der Utility Tree und die klare Strukturierung der Qualitätsattribute helfen, die Anforderungen und Prioritäten gut nachzuvollziehen. 

GaLiGrü 😊
Paula, Safae, Samuel, Valentin, Jonas
